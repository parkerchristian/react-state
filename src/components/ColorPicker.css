.ColorPicker {
  button {
    border: solid black;
    margin-right: 2px;
    margin-bottom: 2px;
    color: white;
  }
  .red {
    background: red;
  }
  .yellow {
    background: yellow;
    color: black;
  }
  .blue {
    background: blue;
  }
  div {
    height: 10em;
    width: 10em;
  }
  .black {
    background: black;
  }
}
