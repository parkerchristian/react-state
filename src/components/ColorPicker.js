import React from 'react';
import PropTypes from 'prop-types';
import styles from './ColorPicker.css';

export default function ColorPicker({ selectedColorButton }) {
  return (
    <section className={styles.ColorPicker}>
      <button className={styles.red} onClick={() => selectedColorButton('red')}>RED</button>
      <button className={styles.yellow} onClick={() => selectedColorButton('yellow')}>YELLOW</button>
      <button className={styles.blue} onClick={() => selectedColorButton('blue')}>BLUE</button>
    </section>
  );
}

ColorPicker.propTypes = {
  selectedColorButton: PropTypes.func.isRequired
};
