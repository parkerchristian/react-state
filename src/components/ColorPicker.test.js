import React from 'react';
import { shallow } from 'enzyme';
import ColorPicker from './ColorPicker';

describe('ColorPicker Tests', () => {
  it('renders color picker', () => {
    const wrapper = shallow(<ColorPicker selectedColorButton={jest.fn()}/>);
    expect(wrapper).toMatchSnapshot();
  });
  
  it('selects red on click', () => {
    const selectedColorButton = jest.fn();
    const wrapper = shallow(<ColorPicker selectedColorButton={selectedColorButton} />);
    wrapper.find('button').at(0).simulate('click');
    expect(selectedColorButton).toHaveBeenCalledWith('red');
  });
  it('selects yellow on click', () => {
    const selectedColorButton = jest.fn();
    const wrapper = shallow(<ColorPicker selectedColorButton={selectedColorButton} />);
    wrapper.find('button').at(1).simulate('click');
    expect(selectedColorButton).toHaveBeenCalledWith('yellow');
  });
  it('selects blue on click', () => {
    const selectedColorButton = jest.fn();
    const wrapper = shallow(<ColorPicker selectedColorButton={selectedColorButton} />);
    wrapper.find('button').at(2).simulate('click');
    expect(selectedColorButton).toHaveBeenCalledWith('blue');
  });

});
