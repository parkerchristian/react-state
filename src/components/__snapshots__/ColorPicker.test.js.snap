// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorPicker Tests renders color picker 1`] = `
<section
  className="ColorPicker"
>
  <button
    className="red"
    onClick={[Function]}
  >
    RED
  </button>
  <button
    className="yellow"
    onClick={[Function]}
  >
    YELLOW
  </button>
  <button
    className="blue"
    onClick={[Function]}
  >
    BLUE
  </button>
</section>
`;
